packages/web-components/src/dialog/dialog.styles.ts

```diff
@@ -32,6 +33,7 @@ export const DialogStyles = css`
     left: 0;
     right: 0;
     overflow: auto;
+    z-index: var(--dialog-z-index);
```

Comment:
Thanks @paucortescosta - we'll take a look at this. Perhaps you can provide a bit of a story around this and how it might scale to other controls that similarly have z-index considerations? We've been looking at larger scale z-index story where we might have a semantic ramp that could be either extended or completely replaced to scale across controls. It seems like something that scales would be preferable here - would that benefit your scenario?

With a solution , I'd likely prefer to bring this into a single line here:

```css
z-index: var(--dialog-z-index, 0);
```